c-bindings-gen/src/types.rs

```diff
@@ -997,7 +996,9 @@ impl<'a, 'c: 'a> TypeResolver<'a, 'c> {
 			"lightning::ln::PaymentHash" if is_ref => Some("&::lightning::ln::PaymentHash(unsafe { *"),
 			"lightning::ln::PaymentPreimage" if !is_ref => Some("::lightning::ln::PaymentPreimage("),
 			"lightning::ln::PaymentPreimage" if is_ref => Some("&::lightning::ln::PaymentPreimage(unsafe { *"),
-			"lightning::ln::PaymentSecret" => Some("::lightning::ln::PaymentSecret("),
+			"lightning::ln::PaymentSecret" if !is_ref => Some("::lightning::ln::PaymentSecret("),
```

Comment:
Was adding `if !is_ref` for `PaymentSecret` intentional?

Reply:
Yes, this only works for the `!is_ref` version, `is_ref` would just hit a compile error.